### PR TITLE
Create session, load graph, and save tensors in ModelWrapper.

### PR DESCRIPTION
**Pull request by sherryy**
When model_path is specified, a session is created for the base
ModelWrapper. The graph is loaded from model_path by trying
a few different graph format. Tensors specified in node_dict
are saved to the corresponding ends and bottleneck_tensors dict.

### CODE REVIEW

File: tcav/model.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import numpy as np
 import six
 import tensorflow as tf
+from google.protobuf import text_format
```
Comment (jameswex) [inline tcav/model.py]: need to add pip dep on protobuf for this, unless we already have it

File: tcav/model.py [hunk 2]
```diff
@@ -45,6 +59,74 @@ def __init__(self):
...
+    The 'ends' and 'bottlenecks_tensors' dictionary should map to tensors
+    with the according name.
+    """
+    self.bottlenecks_tensors = {}
+    self.ends = {}
+    for k, v in node_dict.iteritems():
+      if self.import_prefix:
+        v = 'import/' + v
+      tensor = self.sess.graph.get_operation_by_name(v.strip(':0')).outputs[0]
+      if k == 'input' or k == 'prediction':
```
Comment (jameswex) [inline tcav/model.py]: do we document how people should use input and prediction in their node dict?

File: tcav/model.py [hunk 3]
```diff
@@ -45,6 +59,74 @@ def __init__(self):
...
+    """ Find tensors from the graph by their names.
+
+    Depending on how the model is loaded, tensors in the graph
+    may or may not have 'import/' prefix added to every tensor name.
+    The 'ends' and 'bottlenecks_tensors' dictionary should map to tensors
+    with the according name.
+    """
+    self.bottlenecks_tensors = {}
+    self.ends = {}
+    for k, v in node_dict.iteritems():
```
Comment (jameswex) [inline tcav/model.py]: what if someone gives node dict that includes import/ prefixes?

File: tcav/model_test.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+class ModelTest_model(ModelWrapper):
+  """A mock model of model class for ModelTest class."""
+
+  def __init__(self, model_path=None, node_dict=None):
+    super(ModelTest_model, self).__init__(
+        model_path=model_path, node_dict=node_dict)
+
+  def label_to_id(self, target_class):
+    return 0
```
Comment (jameswex) [inline tcav/model_test.py]: can we have our base class implement a naive version of this for simple models, with others needing to override it? can just return label as id and vice versa (forcing to numbers if necessary)?

File: tcav/model.py (hunk 1 above)
Comment (sherryy) [reply tcav/model.py]: Added it. I also tested it locally using "pip install protobuf" after uninstalled the protobuf dependency, and made sure things still work. Not sure why I had the protobuf dependency initially when I was writing the test.

File: tcav/model.py (hunk 2 above)
Comment (sherryy) [reply tcav/model.py]: Added this info in the doc string of __init__ of ModelWrapper.

File: tcav/model.py (hunk 3 above)
Comment (sherryy) [reply tcav/model.py]: I tried that and 'import/' will still be added to every tensor's name. This is the default behavior of import_graph_def. Added to the comment.

File: tcav/model_test.py (hunk 4 above)
Comment (sherryy) [reply tcav/model_test.py]: If user pass a string 'zebra', forcing it into a number in label_to_id might not make sense. I am defaulting label_to_id to return 0 with logging warnings. I am also defaulting id_to_label to cast idx into a string. id_to_label is not used for now though. Might be useful to keep it around.

**Pull request merged**